### PR TITLE
Updates to CUDA and TensorRT dockerfiles for v0.5.0

### PR DESCRIPTION
**Pull request by vinitra**
**Description**: This PR updates the CUDA dockerfile base image to use CUDA 10.1 and TensorRT dockerfile to use version 19.06. 

**Motivation and Context**
- This way we align with our CI and stay consistent with our newest MCR image updates.
Also adds clear documentation of tags provided to the README.

- Relevant to updates made while addressing #1696

### CODE REVIEW

File: dockerfiles/README.md [hunk 1]
```diff
@@ -8,6 +10,21 @@
 - [OpenVINO](Dockerfile.openvino)
 - [ONNX Runtime Server](Dockerfile.server)
 
+#### Published Microsoft Container Registry (MCR) Images
+
+Use `docker pull` with any of the images and tags below to pull an image and try for yourself. Note that the build from source (CPU), CUDA, and TensorRT images include additional dependencies like miniconda for compatibility with AzureML image deployment.
+
+**Example**: Run `docker pull mcr.microsoft.com/azureml/onnxruntime:latest-cuda` to pull the latest released docker image with ONNX Runtime GPU, CUDA, and CUDNN support.
+
+| Build Flavor      | Base Image                            | ONNX Runtime v0.4.0     | ONNX Runtime v0.5.0     | Latest           |
```
Comment (manashgoswami) [inline dockerfiles/README.md]: Col 3 & 4 should be called "tags" for ORT v0.4.0.

Comment (vinitra) [reply dockerfiles/README.md]: Yes! Addressed in latest update.

**Comment by jywu-msft** [comment]
> FYI, we will revert it to 10.0.

Our thinking here was that AML has necessary drivers for CUDA 10.1,
so we can just go ahead and use 10.1 for the docker image deployed to AML.

Also, The TensorRT base image from Nvidia is using CUDA 10.1

**Comment by snnn** [comment]
> > FYI, we will revert it to 10.0.
> 
> Our thinking here was that AML has necessary drivers for CUDA 10.1,
> so we can just go ahead and use 10.1 for the docker image deployed to AML.
> 
> Also, The TensorRT base image from Nvidia is using CUDA 10.1

Fine.  I'll ask Faith what to do. @faxu

**Comment by vinitra** [comment]
quick bump on this PR. @jywu-msft, @faxu, @snnn, @manashgoswami -- any additional input? This is ready to be merged from my side.

File: dockerfiles/Dockerfile.cuda [hunk 2]
```diff
@@ -17,10 +17,9 @@ RUN apt-get update &&\
 WORKDIR /code
 ENV PATH /usr/local/nvidia/bin:/usr/local/cuda/bin:/code/cmake-3.14.3-Linux-x86_64/bin:/opt/miniconda/bin:${PATH}
 
-# Prepare onnxruntime repository & build onnxruntime with TensorRT
-ADD scripts /tmp/scripts
-RUN /bin/sh /tmp/scripts/install_common_deps.sh && \
-    git clone --single-branch --branch ${ONNXRUNTIME_SERVER_BRANCH} --recursive ${ONNXRUNTIME_REPO} onnxruntime &&\
+# Prepare onnxruntime repository & build onnxruntime with CUDA
+RUN git clone --single-branch --branch ${ONNXRUNTIME_SERVER_BRANCH} --recursive ${ONNXRUNTIME_REPO} onnxruntime &&\
+    /bin/sh onnxruntime/dockerfiles/install_common_deps.sh &&\
```
Comment (manashgoswami) [inline dockerfiles/Dockerfile.cuda]: is this ...dockerfiles/**scripts**/install_common_deps.sh?

File: dockerfiles/README.md [hunk 3]
```diff
@@ -8,6 +10,21 @@
...
+|-------------------|---------------------------------------|--------------------------------------------------|------------------|
+| Source (CPU)      | mcr.microsoft.com/azureml/onnxruntime | :v0.4.0, :v0.5.0                                 | :latest          |
+| CUDA (GPU)        | mcr.microsoft.com/azureml/onnxruntime | :v0.4.0-cuda10.0-cudnn7, :v0.5.0-cuda10.1-cudnn7 | :latest-cuda     |
+| TensorRT (x86)    | mcr.microsoft.com/azureml/onnxruntime | :v0.4.0-tensorrt19.03, :v0.5.0-tensorrt19.06     | :latest-tensorrt |
+| OpenVino (VAD-M)  | mcr.microsoft.com/azureml/onnxruntime | TBA                                              | TBA              |
+| OpenVino (MYRIAD) | mcr.microsoft.com/azureml/onnxruntime | TBA                                              | TBA              |
+| Server            | mcr.microsoft.com/onnxruntime/server  | :v0.4.0, :v0.5.0                                 | :latest          |
+
 **Preparation step:** download `scripts` to your local folder before running the `docker build` command for any of the options below.
```
Comment (manashgoswami) [inline dockerfiles/README.md]: This "preparation step" is not required if line #22 in the dockerfile executes the script after `git clone`

Comment (vinitra) [reply dockerfiles/README.md]: updated

File: dockerfiles/Dockerfile.cuda (hunk 2 above)
Comment (vinitra) [reply dockerfiles/Dockerfile.cuda]: Updated! Looks like it got mixed up in the merging process.

**Pull request merged**